test/test_metadata.py

```diff
@@ -67,3 +76,235 @@ def test_good() -> None:
+            inv_metric=[[1.0, 0.0], [0.0, 1.0]],  # type: ignore
+        )
+        assert isinstance(metric.inv_metric, np.ndarray)
+
+    def test_stepsize_positive(self) -> None:
+        """Test valid positive stepsize"""
+        metric = MetricInfo(
+            stepsize=0.5,
+            metric_type="diag_e",
+            inv_metric=[1.0],  # type: ignore
+        )
+        assert metric.stepsize == 0.5
+
+    def test_stepsize_nan_allowed(self) -> None:
+        """Test that NaN stepsize is allowed"""
+        metric = MetricInfo(
+            stepsize=math.nan,
+            metric_type="diag_e",
+            inv_metric=[1.0],  # type: ignore
+        )
+        assert math.isnan(metric.stepsize)
+
+    def test_stepsize_zero_raises_error(self) -> None:
+        """Test that zero stepsize raises ValueError"""
+        with pytest.raises(ValidationError) as exc_info:
+            MetricInfo(
+                stepsize=0.0,
+                metric_type="diag_e",
+                inv_metric=[1.0],  # type: ignore
+            )
+        assert "stepsize must be greater than 0 or NaN" in str(exc_info.value)
+
+    def test_stepsize_negative_raises_error(self) -> None:
+        """Test that negative stepsize raises ValueError"""
+        with pytest.raises(ValidationError) as exc_info:
+            MetricInfo(
+                stepsize=-0.5,
+                metric_type="diag_e",
+                inv_metric=[1.0],  # type: ignore
+            )
+        assert "stepsize must be greater than 0 or NaN" in str(exc_info.value)
+
+    def test_diag_e_with_2d_array_raises_error(self) -> None:
+        """Test that diag_e with 2D array raises ValueError"""
+        with pytest.raises(ValidationError) as exc_info:
+            MetricInfo(
+                stepsize=0.5,
+                metric_type="diag_e",
+                inv_metric=[[1.0, 2.0]],  # type: ignore
+            )
+        assert "inv_metric must be 1D for diag_e and unit_e" in str(
+            exc_info.value
+        )
+
+    def test_unit_e_with_2d_array_raises_error(self) -> None:
+        """Test that unit_e with 2D array raises ValueError"""
+        with pytest.raises(ValidationError) as exc_info:
+            MetricInfo(
+                stepsize=0.5,
+                metric_type="unit_e",
+                inv_metric=[[1.0], [1.0]],  # type: ignore
+            )
+        assert "inv_metric must be 1D for diag_e and unit_e" in str(
+            exc_info.value
+        )
+
+    def test_dense_e_with_1d_array_raises_error(self) -> None:
+        """Test that dense_e with 1D array raises ValueError"""
+        with pytest.raises(ValidationError) as exc_info:
+            MetricInfo(
+                stepsize=0.5,
+                metric_type="dense_e",
+                inv_metric=[1.0, 2.0],  # type: ignore
+            )
+        assert "Dense inv_metric must be 2D" in str(exc_info.value)
+
+    def test_dense_e_non_square_raises_error(self) -> None:
```

Comment:
I might be missing it, but is there a test for an invalid metric_type?

Reply:
Just added some!